### ISSUE TITLE
pgvector support

### ISSUE THREAD

**Issue by msullivan** [edgedb/edgedb#5478]
We want to be able to support storing vectors from https://github.com/pgvector/pgvector

This probably means
- [ ] Make our extension system powerful enough to specify it as an optional extension package
- [ ] Write an extension package for pgvector
- [ ] Figure out how to package it for when running edgedb-managed postgres

**Comment by msullivan** [comment]
I went and hacked up an initial take on enough of an extension packaging system to probably build this.

I also did a little bit of investigating into pgvector, so here are my very basic notes:
 * The pgvector build system uses `pg_config` to find what to build against and where to install, so I set `PATH=$HOME/src/e/edgedb/build/postgres/install/bin:$PATH` and then just ran `make && make install` and it installed into my edgedb dev build of postgres
 * Probably for dev builds we will just want to add pgvector as a git submodule and then build it as described above. It's under the PostgreSQL license so we can bundle it without any trouble
 * The examples in their README all give a vector length in the types (`vector(3)`, etc`), but just using `vector` without a length works just fine and so we should do that instead
 * By default, `to_json` just takes the text representation of a value and makes it a json string. In my extension system PR, I also made sure to make it so that giving a custom JSON cast works properly and that json output mode works with it, so I think we should consider having the JSON representation be a json array of numbers
